src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *                     username:
+ *                       type: string
+ *                       description: The username of the post creator.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 err:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.post('/', auth, postControl, uploadPost);
+
+/**
+ * @swagger
+ * /posts/{postID}/like:
+ *   post:
+ *     tags:
+ *       - Posts operations
+ *     summary: Like or unlike a post , Must be connected as user
+ *     description: Allows a user to like or unlike a post. If the user likes the post, the post's likes are updated, and the post is added to the user's wishlist if not already present. If the user unlikes the post, the like is removed and the post is removed from the user's wishlist if present.
+ *     security:
+ *       - BearerAuth: []  # Indicates that the endpoint requires a bearer token for authentication
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post to be liked or unliked.
+ *     responses:
+ *       201:
+ *         description: Success message with updated post likes and status.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 posts:
+ *                   type: array
+ *                   items:
+ *                     type: string
+ *                     description: List of user IDs who liked the post.
+ *                 post:
+ *                   type: object
+ *                   properties:
+ *                     _id:
+ *                       type: string
+ *                       description: The unique identifier of the post.
+ *                     likes:
+ *                       type: array
+ *                       items:
+ *                         type: string
+ *                         description: List of user IDs who liked the post.
+ *                 msg:
+ *                   type: string
+ *                   description: Status message indicating whether the post was liked or unliked.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 message:
+ *                   type: string
+ *                   description: Error message.
+ */
 router.post('/likePost/:postID', auth, likePost);
+
+/**
+ * @swagger
+ * /posts/singleImgDel/{postID}/{imgID}:
+ *   delete:
+ *     tags:
+ *       - Posts operations
+ *     summary: Delete an image from a post , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"